tests/src/core/CommandTest.cpp

```diff
@@ -0,0 +1,124 @@
+/*
+ * CommandTest.cpp
+ *
+ * Copyright (c) 2025 szeli1 <TODO/at/gmail/dot/com>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+
+#include <QtTest>
+#include "Command.h"
+#include "CommandStack.h"
+
+
+class CommandTestObject
+{
+public:
+	CommandTestObject(lmms::CommandStack& stack)
+		: m_value{0.0f}
+		, m_boolean{false}
+		, m_setValueA{new lmms::ParamCommandLambda{stack,
+			[this](float newVal, float& oldVal) { oldVal = m_value; setTestValue(newVal); },
+			[this](float newVal, float oldVal) { setTestValue(oldVal); }, 0.0f}}
+		, m_setValueB{stack, *this, &CommandTestObject::getTestValue, &CommandTestObject::setTestValue, nullptr} // getter
+		, m_setValueC{stack, *this, nullptr, &CommandTestObject::setTestValue, &m_value} // reference
+		, m_doActionA{new lmms::CommandLambda{stack,
+			[this]() { doAction(); },
+			[this]() { undoAction(); }}}
+		, m_doActionB{stack, *this, &CommandTestObject::doAction, &CommandTestObject::undoAction}
+	{}
+	~CommandTestObject()
+	{
+		delete m_setValueA;
+		delete m_doActionA;
+	}
+	float m_value;
+	bool m_boolean;
+
+	void doAction() { m_boolean = true; };
+	void undoAction() { m_boolean = false; };
+
+	void setTestValue(float value) { m_value = value; };
+	float getTestValue() { return m_value; };
+
+	lmms::ParamCommand<float>* m_setValueA;
+	lmms::ParamCommandFnPtr<float, CommandTestObject, void> m_setValueB;
+	lmms::ParamCommandFnPtr<float, CommandTestObject, void> m_setValueC;
+	lmms::TypelessCommand* m_doActionA;
+	lmms::CommandFnPtr<CommandTestObject, void, void> m_doActionB;
+};
+
+class CommandTest : public QObject
+{
+	Q_OBJECT
+private slots: // tests
+
+	void CommandTests()
+	{
+		using namespace lmms;
+
+		CommandStack stack;
+		CommandTestObject obj{stack};
+		
+		QVERIFY(obj.m_value == 0.0f); // these should be the default values
```

Comment:
```suggestion
		QCOMPARE(obj.m_value, 0.0f); // these should be the default values
```
`QCOMPARE` is better than `QVERIFY` since if it fails, it will print out what the actual value of `obj.m_value` was and what it expected it to be, but `QVERIFY` will only tell you that it failed.